opengemini/query_test.go

```diff
@@ -77,7 +77,183 @@ func TestQueryWithMsgPack(t *testing.T) {
+	})
+
+	// create a test database with rand suffix
+	database := randomDatabaseName()
+	err := c.CreateDatabase(database)
+	assert.Nil(t, err)
+
+	// delete test database before exit test case
+	defer func() {
+		err := c.DropDatabase(database)
+		assert.Nil(t, err)
+	}()
+
+	testMeasurement := randomMeasurement()
+	p := &Point{}
+	p.Measurement = testMeasurement
+	p.AddField("TestField", 123)
+	p.Time = time.Now()
+
+	err = c.WritePoint(database, p, func(err error) {
+		assert.Nil(t, err)
+	})
+	assert.Nil(t, err)
+
+	time.Sleep(time.Second * 5)
+
+	PrecisionTimestampLength := make(map[Precision]int64)
+	PrecisionTimestampLength[PrecisionNanosecond] = 19
+	PrecisionTimestampLength[PrecisionMicrosecond] = 16
+	PrecisionTimestampLength[PrecisionMillisecond] = 13
+	PrecisionTimestampLength[PrecisionSecond] = 10
+	PrecisionTimestampLength[PrecisionMinute] = 8
+	PrecisionTimestampLength[PrecisionHour] = 6
+
+	// check whether write success
+	for precision, length := range PrecisionTimestampLength {
+		q := Query{
+			Database:  database,
+			Command:   "select * from " + testMeasurement,
+			Precision: precision,
+		}
+		result, err := c.Query(q)
+		assert.Nil(t, err)
+		v, err := convertToInt64(result.Results[0].Series[0].Values[0][0])
+		if err != nil {
+			t.Fatalf("conversion error: %v", err)
+		}
+		assert.Equal(t, length, getTimestampLength(v))
+	}
+}
+
+func TestQueryWithZSTD(t *testing.T) {
+	c := testNewClient(t, &Config{
+		Addresses: []addr.Address{{
+			Host: "localhost",
+			Port: 8086,
+		}},
+		CompressMethod: CompressMethodZstd,
+	})
+
+	// create a test database with rand suffix
+	database := randomDatabaseName()
+	err := c.CreateDatabase(database)
+	assert.Nil(t, err)
+
+	// delete test database before exit test case
+	defer func() {
+		err := c.DropDatabase(database)
+		assert.Nil(t, err)
+	}()
+
+	testMeasurement := randomMeasurement()
+	p := &Point{}
+	p.Measurement = testMeasurement
+	p.AddField("TestField", 123)
+	p.Time = time.Now()
+
+	err = c.WritePoint(database, p, func(err error) {
+		assert.Nil(t, err)
+	})
+	assert.Nil(t, err)
+
+	time.Sleep(time.Second * 5)
+
+	PrecisionTimestampLength := make(map[Precision]int64)
+	PrecisionTimestampLength[PrecisionNanosecond] = 19
+	PrecisionTimestampLength[PrecisionMicrosecond] = 16
+	PrecisionTimestampLength[PrecisionMillisecond] = 13
+	PrecisionTimestampLength[PrecisionSecond] = 10
+	PrecisionTimestampLength[PrecisionMinute] = 8
+	PrecisionTimestampLength[PrecisionHour] = 6
+
+	// check whether write success
+	for precision, length := range PrecisionTimestampLength {
+		q := Query{
+			Database:  database,
+			Command:   "select * from " + testMeasurement,
+			Precision: precision,
+		}
+		result, err := c.Query(q)
+		assert.Nil(t, err)
+		v, err := convertToInt64(result.Results[0].Series[0].Values[0][0])
+		if err != nil {
+			t.Fatalf("conversion error: %v", err)
+		}
+		assert.Equal(t, length, getTimestampLength(v))
+	}
+}
+
+// func TestQueryWithGZIPAndMsgPack(t *testing.T) {
```

Comment:
should enable when issue https://github.com/openGemini/opengemini-client-go/issues/200 fix